backend/question-service/controllers/question.controller.js

```diff
@@ -0,0 +1,117 @@
+const db = require('../models');
+const Question = db.questions;
+
+exports.create = (req, res) => {
+  if (!req.body.questionId
+    || !req.body.questionTitle
+    || !req.body.questionDescription
+    || !req.body.questionCategory
+    || !req.body.questionComplexity) {
+    res.status(400).send({message: 'Question details missing.'});
+    return;
+  }
+
+  const question = new Question({
+    questionId: req.body.questionId,
+    questionTitle: req.body.questionTitle,
+    questionDescription: req.body.questionDescription,
+    questionCategory: req.body.questionCategory.split(',').map(category => category.trim()),
+    questionComplexity: req.body.questionComplexity
+  });
+
+  question
+    .save(question)
+    .then(data => {
+      res.send(data);
+    })
+    .catch(err => {
+      res.status(500).send({
+        message: err.message || 'An error occured while creating the Question.'
+      })
+    });
+  
+  return;
+};
+
+exports.findOne = (req, res) => {
+  const questionId = req.params.questionId;
+  var condition = questionId ? { questionId: questionId } : {};
+
+  Question.findOne(condition)
+    .then(data => {
+      res.send(data);
+      return;
+    })
+    .catch(err => {
+      res.status(500).send({
+        message: err.message || `An error occured while retrieving the question with questionId ${questionId}.`
+      })
+      return;
+    });
+};
+
+exports.update = (req, res) => {
+  if (!req.body) {
+    return res.status(400).send({
+      message: 'Details required to update.'
+    });
+  }
+
+  const questionId = req.params.questionId;
+  var condition = questionId ? { questionId: questionId } : {};
+
+  Question.findOneAndUpdate(condition, req.body, { new: true })
+    .then(data => {
+      if (!data) {
+        res.status(404).send({
```

Comment:
404 is not found right? Consider using error 400 instead.

Reply:
I think it's 404 too since the request was valid/correct but the id is missing? That's what I understand from the article below at least. 
https://www.ibm.com/docs/en/alm/1.3.0?topic=reference-api-http-status-codes